<a href="https://colab.research.google.com/github/leonardoub/SCRIPT_PALERMO/blob/master/prova_visualize_QDA.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#Visualize using QDA (Quadratic Discriminant Analysis) dati non standardizzati

In [0]:
from sklearn.discriminant_analysis import QuadraticDiscriminantAnalysis
import seaborn as sns

In [0]:
train_data = df_train.drop(['Histology', 'Surv_time_months', 'OS', 'Deadstatus_event','Overall_Stage'], axis=1)

In [0]:
test_data = df_test.drop(['Histology', 'Surv_time_months', 'OS', 'Deadstatus_event','Overall_Stage'], axis=1)

In [0]:
train_labels = df_train.Histology

In [0]:
test_labels = df_test.Histology

In [0]:
qda = QuadraticDiscriminantAnalysis()
qda_result = qda.fit(train_data, train_labels)
train_data_qda = qda.predict(train_data)

/usr/local/lib/python3.6/dist-packages/sklearn/discriminant_analysis.py:691: UserWarning: Variables are collinear
  warnings.warn("Variables are collinear")


In [0]:
train_data_qda.shape

(131,)

In [0]:
type(qda_result)

sklearn.discriminant_analysis.QuadraticDiscriminantAnalysis

In [0]:
df

lda_one   lda_two
0   -3.735330  5.539015
1   -2.972100  3.896330
2   -4.492177  4.515088
3   -3.994613  4.380586
4   -2.207364  4.471812
5   -4.445929  4.405227
6   -2.847913  5.043523
7   -4.522311  5.011720
8   -4.777269  5.723481
9   -3.607391  4.414453
10  -2.468157  2.416140
11  -3.646422  3.868567
12  -3.801610  3.479738
13  -4.548264  5.540532
14  -3.377001  3.110961
15  -3.982921  5.089251
16  -2.226704 -1.184783
17  -2.042138 -1.200551
18  -1.139653 -1.711031
19  -1.648478 -1.485294
20  -2.006063 -2.299054
21  -2.216541 -1.364580
22  -1.444258 -1.554363
23  -2.187529 -1.903289
24  -1.867875 -2.282423
25  -2.265980 -2.517977
26  -2.829232 -0.491450
27  -1.431311 -0.455642
28  -1.652268 -2.130679
29  -0.054812  0.281458
30  -0.508007 -1.255800
31  -1.625618 -1.972012
32  -2.563817 -3.480215
33  -2.129496 -1.760648
34  -0.656685 -0.450078
35  -0.613257 -2.168432
36  -1.811496  0.082480
37  -1.328539 -2.054160
38  -3.604222  0.091668
39  -1.448284 -0.729829
40  -2.545470 -1.120844
41  -2.227532 -2.858677
42  -2.932452  0.009480
43  -1.335501 -2.987213
44  -1.733442 -1.358058
45  -2.530635 -3.142995
46  -1.815313 -2.152367
47  -2.328400 -1.141314
48  -1.623191 -1.657286
49  -2.345370 -2.411171
50   0.099184 -2.516558
51  -0.834825 -0.379983
52  -1.319675 -2.683886
53  -2.395952 -0.344791
54  -1.961620 -2.119340
55  -2.985948 -0.505973
56  -1.460413 -1.536139
57  -0.682022  0.423369
58  -3.066090 -2.927663
59  -2.094733 -0.798484
60  -1.132084 -2.341397
61   0.319280 -1.475689
62  -2.402990 -1.948709
63  -4.045318 -2.986662
64  -1.096936 -3.079413
65  -1.004272  0.135345
66  -0.306977 -0.805607
67  -2.710284 -3.321326
68  -0.624241 -1.065434
69  -0.089931 -2.066037
70  -0.397531 -3.400512
71  -0.837009 -2.699400
72  -2.498101 -2.063629
73  -1.863668 -1.961060
74  -2.885402 -2.857257
75  -1.986429 -0.317835
76  -1.995030 -2.102447
77   1.688456  2.054007
78   3.048100  1.342450
79   2.062952  1.216852
80   2.917798 -0.759574
81   3.142641  0.238385
82   1.173154  1.006105
83   3.494747  0.615579
84   2.287602  0.764050
85   1.949361 -0.501264
86   1.243833 -0.942038
87   2.232677  0.971572
88   2.207048  1.322143
89   2.768567  0.875080
90   2.580384  1.128180
91   1.035791  0.979126
92   3.330065  0.801056
93   3.610590  0.306195
94   4.782017  0.333010
95   2.804247  0.705635
96   2.679591  0.443527
97   3.014223  1.448603
98   5.166191 -0.854692
99   3.992558  0.136697
100  4.937522  1.801495
101  3.905312  0.266908
102  2.266282 -0.238956
103  3.709152 -1.420607
104  3.650251 -0.047334
105  2.883895  0.350723
106  4.561816  1.612537
107  4.369875  1.609677
108  3.537812 -0.410176
109  2.877009  0.932515
110  5.247092  3.593688
111  1.968401  1.426520
112  2.154401  0.866421
113  0.627055 -0.916787
114  3.142489  0.115672
115  3.563864 -0.143705
116  0.526274  1.058014
117  2.797983  1.201740
118  1.678241  0.599673
119  2.406749  0.200220
120  4.792909  1.282634
121  4.709860 -1.190552
122  3.943119  0.470754
123  3.886706  0.607175
124  3.571755 -0.510090
125  4.037821  0.432083
126  2.270244  0.681552
127  2.862014  1.671285
128  4.186723 -2.234604
129  3.013112  0.917586
130  3.107027  1.440474

In [0]:
print('Explained variation per principal component: {}'.format(lda.explained_variance_ratio_))

Explained variation per principal component: [0.6450461 0.3549539]


La prima componente ha il 65% della varianza dell'intero dataset, la seconda il 35%